src/Atc.Rest.Client/Builder/MessageRequestBuilder.cs

```diff
@@ -56,56 +56,48 @@ public IMessageRequestBuilder WithBody<TBody>(TBody body)
             return this;
         }
 
-        public IMessageRequestBuilder WithPathParameter(string name, string value)
+        public IMessageRequestBuilder WithPathParameter(string name, object? value)
         {
             if (string.IsNullOrWhiteSpace(name))
             {
                 throw new ArgumentException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
             }
 
-            if (string.IsNullOrWhiteSpace(value))
+            if (value is null || string.IsNullOrWhiteSpace(value.ToString()))
             {
                 throw new ArgumentException($"'{nameof(value)}' cannot be null or whitespace", nameof(value));
             }
 
-            pathMapper[name] = value;
+            pathMapper[name] = value.ToString();
 
             return this;
         }
 
-        public IMessageRequestBuilder WithPathParameter<T>(string name, T value)
-            where T : struct
-            => WithPathParameter(name, value.ToString());
-
-        public IMessageRequestBuilder WithHeaderParameter(string name, string value)
+        public IMessageRequestBuilder WithHeaderParameter(string name, object? value)
         {
             if (string.IsNullOrWhiteSpace(name))
             {
                 throw new ArgumentException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
```

Comment:
```suggestion
                throw new ArgumentNullException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
```

Pr. the interface docs, this should be the correct exception to throw here.